scripts/generate-framework-agreement-counterpart-signature-pages.py

```diff
@@ -97,15 +99,17 @@
     supplier_ids = get_supplier_ids_from_args(args)
     html_dir = tempfile.mkdtemp()
 
-    records = find_suppliers_with_details_and_draft_service_counts(client, framework_slug, supplier_ids)
+    records = find_suppliers_with_signed_framework_agreements(client, framework_slug, supplier_ids)
     records = list(records)
-    for record in records:
-        if (framework_supports_e_signature(framework) and
-                client.get_framework_agreement(record['agreementId'])['agreement']['status'] == 'signed'):
-            # E-signatures are automatically approved.
-            client.approve_agreement_for_countersignature(record['agreementId'],
-                                                          "automated countersignature script",
-                                                          AUTOMATED_COUNTERSIGNING_USER_ID)
+    if framework_supports_e_signature(framework):
+        for record in records:
+            if not record["countersignedAt"]:
+                # E-signatures are automatically approved.
+                countersigned_agreement = client.approve_agreement_for_countersignature(
+                                              record['agreementId'],
+                                              "automated countersignature script",
+                                              AUTOMATED_COUNTERSIGNING_USER_ID)["agreement"]
+                record["countersignedAt"] = countersigned_agreement["countersignedAgreementReturnedAt"]
```

Comment:
Does this edit get sent back to the API?

Reply:
this is setting in the `record` what the API has stored and just returned in: https://github.com/alphagov/digitalmarketplace-scripts/blob/94202543a6e883e609e84973449b4871c90ab0fa/scripts/generate-framework-agreement-counterpart-signature-pages.py#L108

Reply:
Am I misunderstanding how this works? What is `record`?

If `record` is just a dictionary, how does our edit to it get into the database?

Reply:
`record` is a kind of pimped list of suppliers, which gets augmented with extra details required for the document i.e. in https://github.com/alphagov/digitalmarketplace-scripts/blob/94202543a6e883e609e84973449b4871c90ab0fa/dmscripts/helpers/framework_helpers.py#L85-L96
Because this approval is an additional API call that happens later on in the process and conditionally, we have to update the record object at that point with the `countersignedAt` value, so that subsequent check in: https://github.com/alphagov/digitalmarketplace-scripts/blob/27288765a66bcb50c568658c6d14a91489546fbe/dmscripts/generate_framework_agreement_signature_pages.py#L104 passes.

Does that make more sense?